Data race condition failure in the tests

How to reproduce:
```console
$ make test
[....]
Makefile:115: recipe for target 'test-unit' failed
```

Issue shown in the test output:
```console
=== RUN   TestResourceSortOrder
==================
WARNING: DATA RACE
Write at 0x00c000366978 by goroutine 57:
  k8s.io/helm/vendor/k8s.io/client-go/rest/fake.(*RESTClient).Do()
      /home/usr1/go/src/k8s.io/helm/vendor/k8s.io/client-go/rest/fake/fake.go:113 +0x69
  k8s.io/helm/vendor/k8s.io/client-go/rest.(*Request).request()
      /home/usr1/go/src/k8s.io/helm/vendor/k8s.io/client-go/rest/request.go:737 +0x769
  k8s.io/helm/vendor/k8s.io/client-go/rest.(*Request).Do()
      /home/usr1/go/src/k8s.io/helm/vendor/k8s.io/client-go/rest/request.go:809 +0xd5
  k8s.io/helm/vendor/k8s.io/cli-runtime/pkg/resource.(*Helper).Get()
      /home/usr1/go/src/k8s.io/helm/vendor/k8s.io/cli-runtime/pkg/resource/helper.go:61 +0x131
  k8s.io/helm/vendor/k8s.io/cli-runtime/pkg/resource.(*Info).Get()
      /home/usr1/go/src/k8s.io/helm/vendor/k8s.io/cli-runtime/pkg/resource/visitor.go:100 +0x27a
  k8s.io/helm/pkg/kube.(*Client).Get.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:231 +0x241
  k8s.io/helm/pkg/kube.batchPerform.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:643 +0x41

Previous write at 0x00c000366978 by goroutine 59:
  k8s.io/helm/vendor/k8s.io/client-go/rest/fake.(*RESTClient).Do()
      /home/usr1/go/src/k8s.io/helm/vendor/k8s.io/client-go/rest/fake/fake.go:113 +0x69
  k8s.io/helm/vendor/k8s.io/client-go/rest.(*Request).request()
      /home/usr1/go/src/k8s.io/helm/vendor/k8s.io/client-go/rest/request.go:737 +0x769
  k8s.io/helm/vendor/k8s.io/client-go/rest.(*Request).Do()
      /home/usr1/go/src/k8s.io/helm/vendor/k8s.io/client-go/rest/request.go:809 +0xd5
  k8s.io/helm/vendor/k8s.io/cli-runtime/pkg/resource.(*Helper).Get()
      /home/usr1/go/src/k8s.io/helm/vendor/k8s.io/cli-runtime/pkg/resource/helper.go:61 +0x131
  k8s.io/helm/vendor/k8s.io/cli-runtime/pkg/resource.(*Info).Get()
      /home/usr1/go/src/k8s.io/helm/vendor/k8s.io/cli-runtime/pkg/resource/visitor.go:100 +0x27a
  k8s.io/helm/pkg/kube.(*Client).Get.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:231 +0x241
  k8s.io/helm/pkg/kube.batchPerform.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:643 +0x41

Goroutine 57 (running) created at:
  k8s.io/helm/pkg/kube.batchPerform()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:642 +0xee

Goroutine 59 (finished) created at:
  k8s.io/helm/pkg/kube.batchPerform()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:642 +0xee
==================
==================
WARNING: DATA RACE
Read at 0x00c0005caff0 by goroutine 57:
  runtime.mapaccess1_faststr()
      /usr/local/go/src/runtime/map_faststr.go:12 +0x0
  k8s.io/helm/pkg/kube.(*Client).Get.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:245 +0x73a
  k8s.io/helm/pkg/kube.batchPerform.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:643 +0x41

Previous write at 0x00c0005caff0 by goroutine 59:
  runtime.mapassign_faststr()
usr/local/go/src/runtime/map_faststr.go:202 +0x0
  k8s.io/helm/pkg/kube.(*Client).Get.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:246 +0xb0e
  k8s.io/helm/pkg/kube.batchPerform.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:643 +0x41

Goroutine 57 (running) created at:
  k8s.io/helm/pkg/kube.batchPerform()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:642 +0xee

Goroutine 59 (finished) created at:
  k8s.io/helm/pkg/kube.batchPerform()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:642 +0xee
==================
==================
WARNING: DATA RACE
Read at 0x00c0008f2088 by goroutine 57:
  k8s.io/helm/pkg/kube.(*Client).Get.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:245 +0x750
  k8s.io/helm/pkg/kube.batchPerform.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:643 +0x41

Previous write at 0x00c0008f2088 by goroutine 59:
  k8s.io/helm/pkg/kube.(*Client).Get.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:246 +0xb26
  k8s.io/helm/pkg/kube.batchPerform.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:643 +0x41

Goroutine 57 (running) created at:
  k8s.io/helm/pkg/kube.batchPerform()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:642 +0xee

Goroutine 59 (finished) created at:
  k8s.io/helm/pkg/kube.batchPerform()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:642 +0xee
==================
==================
WARNING: DATA RACE
Write at 0x00c0005cbef0 by goroutine 57:
  runtime.mapassign_faststr()
      /usr/local/go/src/runtime/map_faststr.go:202 +0x0
  k8s.io/helm/pkg/kube.(*Client).Get.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:249 +0x83c
  k8s.io/helm/pkg/kube.batchPerform.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:643 +0x41

Previous write at 0x00c0005cbef0 by goroutine 59:
  runtime.mapassign_faststr()
      /usr/local/go/src/runtime/map_faststr.go:202 +0x0
  k8s.io/helm/pkg/kube.(*Client).Get.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:249 +0x83c
  k8s.io/helm/pkg/kube.batchPerform.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:643 +0x41

Goroutine 57 (running) created at:
  k8s.io/helm/pkg/kube.batchPerform()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:642 +0xee

Goroutine 59 (finished) created at:
  k8s.io/helm/pkg/kube.batchPerform()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:642 +0xee
==================
==================
WARNING: DATA RACE
Read at 0x00c000010070 by goroutine 57:
  k8s.io/helm/pkg/kube.(*Client).Get.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:252 +0x88e
k8s.io/helm/pkg/kube.batchPerform.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:643 +0x41

Previous write at 0x00c000010070 by goroutine 59:
  k8s.io/helm/pkg/kube.(*Client).Get.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:252 +0x8f1
  k8s.io/helm/pkg/kube.batchPerform.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:643 +0x41

Goroutine 57 (running) created at:
  k8s.io/helm/pkg/kube.batchPerform()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:642 +0xee

Goroutine 59 (finished) created at:
  k8s.io/helm/pkg/kube.batchPerform()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:642 +0xee
==================
==================
WARNING: DATA RACE
Write at 0x00c0002e0920 by goroutine 57:
  k8s.io/helm/pkg/kube.(*Client).Get.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:252 +0x922
  k8s.io/helm/pkg/kube.batchPerform.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:643 +0x41

Previous write at 0x00c0002e0920 by goroutine 59:
  k8s.io/helm/pkg/kube.(*Client).Get.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:252 +0x922
  k8s.io/helm/pkg/kube.batchPerform.func1()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:643 +0x41

Goroutine 57 (running) created at:
  k8s.io/helm/pkg/kube.batchPerform()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:642 +0xee

Goroutine 59 (finished) created at:
  k8s.io/helm/pkg/kube.batchPerform()
      /home/usr1/go/src/k8s.io/helm/pkg/kube/client.go:642 +0xee
==================
--- FAIL: TestResourceSortOrder (0.06s)
    client_test.go:469: got request /namespaces/default/pods/beluga GET
    client_test.go:469: got request /namespaces/default/pods/albacore GET
    client_test.go:469: got request /namespaces/default/pods/coral GET
    testing.go:809: race detected during execution of test
```



Output of `helm version`:
Client: &version.Version{SemVer:"v2.14+unreleased", GitCommit:"fb2cbb001986b28dec5303f5acad65390a1a8118", GitTreeState:"dirty"}

Output of `kubectl version`:

Cloud Provider/Platform (AKS, GKE, Minikube etc.):

--- a/pkg/kube/client.go
+++ b/pkg/kube/client.go
@@ -218,6 +218,8 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
 	// Since we don't know what order the objects come in, let's group them by the types and then sort them, so
 	// that when we print them, they come out looking good (headers apply to subgroups, etc.).
 	objs := make(map[string](map[string]runtime.Object))
+	mux := &sync.Mutex{}
+
 	infos, err := c.BuildUnstructured(namespace, reader)
 	if err != nil {
 		return "", err
@@ -227,6 +229,8 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
 
 	missing := []string{}
 	err = perform(infos, func(info *resource.Info) error {
+		mux.Lock()
+		defer mux.Unlock()
 		c.Log("Doing get for %s: %q", info.Mapping.GroupVersionKind.Kind, info.Name)
 		if err := info.Get(); err != nil {
 			c.Log("WARNING: Failed Get for resource %q: %s", info.Name, err)

